source/ingress.go

```diff
@@ -74,10 +74,7 @@ func endpointsFromIngress(ing *v1beta1.Ingress) []*endpoint.Endpoint {
 				endpoints = append(endpoints, endpoint.NewEndpoint(sanitizeHostname(rule.Host), lb.IP))
 			}
 			if lb.Hostname != "" {
-				endpoints = append(endpoints, endpoint.Endpoint{
-					DNSName: sanitizeHostname(rule.Host),
-					Target:  sanitizeHostname(lb.Hostname),
-				})
+				endpoints = append(endpoints, endpoint.NewEndpoint(sanitizeHostname(rule.Host), sanitizeHostname(lb.Hostname)))
```

Comment:
@linki yes, I think this can be included. Though I am not sure if this will have any impact on the logic where we do the `DNSName, Target` string comparisons